docs/user/secure-gateways.md

```diff
@@ -0,0 +1,214 @@
+
+```shell
+curl --version | grep LibreSSL
+curl 7.54.0 (x86_64-apple-darwin17.0) libcurl/7.54.0 LibreSSL/2.0.20 zlib/1.2.11 nghttp2/1.24.0
+```
+
+Create a root certificate and private key to sign certificates:
+
+```shell
+openssl req -x509 -sha256 -nodes -days 365 -newkey rsa:2048 -subj '/O=example Inc./CN=example.com' -keyout example.com.key -out example.com.crt
+```
+
+Create a certificate and a private key for `www.example.com`:
+
+```shell
+openssl req -out www.example.com.csr -newkey rsa:2048 -nodes -keyout www.example.com.key -subj "/CN=www.example.com/O=httpbin organization"
+openssl x509 -req -days 365 -CA example.com.crt -CAkey example.com.key -set_serial 0 -in www.example.com.csr -out www.example.com.crt
+```
+
+Store the cert/key in a Secret:
+
+```shell
+kubectl create secret tls example-cert --key=www.example.com.key --cert=www.example.com.crt
+```
+
+## Testing
+
+### Clusters without External Loadbalancer Support
+
+Port forward to the Envoy service:
+
+```shell
+kubectl -n envoy-gateway-system port-forward service/envoy-default-eg 8043:8443 &
+```
+
+Curl the example app through Envoy proxy:
+
+```shell
+curl -v -HHost:www.example.com --resolve "www.example.com:8043:127.0.0.1" \
+--cacert example.com.crt https://www.example.com:8043/get
+```
+
+You can replace `get` with any of the supported [httpbin methods][httpbin_methods].
+
+### Clusters with External Loadbalancer Support
+
+Get the External IP of the Gateway:
+```shell
+export GATEWAY_HOST=$(kubectl get gateway/eg -o jsonpath='{.status.addresses[0].value}')
+```
+
+Curl the example app through the Gateway:
+
+```shell
+curl -v -HHost:www.example.com --resolve "www.example.com:8443:${GATEWAY_HOST}" \
+--cacert example.com.crt https://www.example.com:8443/get
+```
+
+You can replace `get` with any of the supported [httpbin methods][httpbin_methods].
+
+## Multiple HTTPS Listeners
+
+Due to [Issue 520][], multiple HTTP listeners must use different port numbers. For example:
+
+```shell
+cat <<EOF | kubectl apply -f -
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: eg
+spec:
+  gatewayClassName: eg
+  listeners:
+    - name: http
+      protocol: HTTP
+      port: 8080
+    - name: https
+      protocol: HTTPS
+      port: 8443
+      tls:
+        mode: Terminate
+        certificateRefs:
+          - kind: Secret
+            group: ""
+            name: example-cert
+    - name: https-2
+      protocol: HTTPS
+      port: 8444
+      tls:
+        mode: Terminate
+        certificateRefs:
+          - kind: Secret
+            group: ""
+            name: example-cert-2
+ EOF
+```
+
+Store the previously created cert/key in Secret "example-cert-2":
+
+```shell
+kubectl create secret tls example-cert-2 --key=www.example.com.key --cert=www.example.com.crt
+```
+
+Follow the steps in the [Testing section](#testing) to test connectivity to the backend app through both Gateway
+listeners.
+
+## Cross Namespace Certificate References
```

Comment:
Couple notes here, I'll see if there are existing issues for these in a sec:
- it seems like if you create the ReferenceGrant as the last step, it doesn't take effect until an EG restart or something else kicks the Gateway to re-reconcile
- Once everything's working, if you then delete the ReferenceGrant, the Listener continues to be "Ready" and still serves traffic with the TLS cert that it shouldn't be allowed to use.

Reply:
Sam feedback as https://github.com/envoyproxy/gateway/pull/542/files#r997713038.

Reply:
This is now fixed and doc updated.